### PR TITLE
add computation of dhg/dq inside computecentroidaldynamicsderivatives

### PR DESCRIPTION
**Pull request by proyan**
I extend the forceSet::motionAction to include the case when multiple column vectors are being applied to a single force vector.

Except that, the implementation is easy to understand.

### CODE REVIEW

File: src/algorithm/centroidal-derivatives.hxx [hunk 1]
```diff
@@ -332,14 +342,18 @@ namespace pinocchio
...
       }
+
+      forceSet::motionActions(J_cols, data.oh[i], dHdq_cols);
+      motionSet::inertiaAction<ADDTO>(data.oYcrb[i], dVdq_cols, dHdq_cols);
     }
   }; // struct GetCentroidalDynDerivativesBackwardStep
   
   template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl,
-           typename Matrix6xLike1, typename Matrix6xLike2, typename Matrix6xLike3>
+           typename Matrix6xLike0,typename Matrix6xLike1, typename Matrix6xLike2, typename Matrix6xLike3>
```
Comment (jcarpent) [inline src/algorithm/centroidal-derivatives.hxx]: Missing space here.

File: src/spatial/act-on-set.hpp [hunk 2]
```diff
@@ -64,6 +64,22 @@ namespace pinocchio
...
                              Eigen::MatrixBase<MatRet> const & jF);
+
+    ///
+    /// \brief Action of a set of motions on a force. Motion is
+    ///        represented by a 6xN matrix whose each
+    ///        column represent a spatial motion.
+    ///
+    template<int Op, typename Mat, typename ForceDerived, typename MatRet>
+    static void motionActions(const Eigen::MatrixBase<Mat> & iV,
+                             const ForceDense<ForceDerived> & f,
```
Comment (jcarpent) [inline src/spatial/act-on-set.hpp]: Missing indentation

File: src/algorithm/centroidal-derivatives.hxx (hunk 1 above)
Comment (proyan) [reply src/algorithm/centroidal-derivatives.hxx]: Done.

File: src/spatial/act-on-set.hpp (hunk 2 above)
Comment (proyan) [reply src/spatial/act-on-set.hpp]: Done.

**Comment by proyan** [comment]
It seems that I missed the motionSet::act, it is the same implementation as forceSet::motionAction.
(Which is a bit confusing, since all implementation of motion and force operations is done in the Force class, or in the forceSet namespace. motionSet::act is an outlier here.)

I've removed the commit which added forceset::motionActions.

**Pull request merged**